plugins/modules/usb_facts.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2024, Max Maxopoly <max@dermax.org>
+#
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+---
+module: usb_facts
+short_description: Allows listing information about usb devices
+description:
+  - Allows retrieving information about available usb devices through lsusb
+author:
+  - Max Maxopoly (@maxopoly)
+extends_documentation_fragment:
+  - community.general.attributes
+requirements:
+  - lsusb binary on PATH (usually installed through the package usbutils and preinstalled on many systems)
+attributes:
+  check_mode:
+    support: full
+  diff_mode:
+    support: none
+'''
+
+EXAMPLES = '''
+- name: Get information about USB devices
+  community.general.usb_facts:
+  register: usb_facts
+'''
+
+RETURN = r'''
+ansible_facts:
+  description: Dictionary containing details of connected usb devices
+  returned: always
+  type: complex
+  contains:
+    devices:
+      description: A list of usb devices available
+      returned: always
+      type: list
+      contains:
+        bus:
+          description: The bus the usb device is connected to.
+          returned: always
+          type: str
+          sample: "001"
+        device:
+          description: The device number occupied on the bus
+          returned: always
+          type: str
+          sample: "002"
+        id:
+          description: Id of the usb device
+          returned: always
+          type: str
+          sample: "1d6b:0002"
+        name:
+          description: Human readable name of the device
+          returned: always
+          type: str
+          sample: Linux Foundation 2.0 root hub
+'''
+
+import re
+# Import module snippets.
```

Comment:
this is not really necessary

Reply:
That snuck in along with copy pasting imports, I've removed it.